miden/tests/integration/exec_iters.rs

```diff
@@ -0,0 +1,155 @@
+use super::build_test;
+use processor::VmState;
+use vm_core::{utils::ToElements, Felt};
+
+// EXEC ITER TESTS
+// =================================================================
+#[test]
+fn test_exec_iter() {
+    let source = "proc.foo.1 push.local.0 end begin popw.mem.1 push.17 exec.foo end";
+    let mut init_stack: Vec<u64> = Vec::new();
+    (1..=16).for_each(|i| {
+        init_stack.push(i);
+    });
+    let test = build_test!(source, &init_stack.clone());
+    let traces = test.execute_iter();
+    let fmp = Felt::new(1073741824);
+    let mem = vec![(1_u64, slice_to_word(&[13, 14, 15, 16]))];
+    let expected_states = vec![
+        VmState {
+            clk: 0,
+            stack: [16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1].to_elements(),
+            fmp: fmp,
+            memory: Vec::new(),
+        },
+        VmState {
+            clk: 1,
+            stack: [16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 1].to_elements(),
+            fmp: fmp,
+            memory: Vec::new(),
+        },
+        VmState {
+            clk: 2,
+            stack: [1, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2].to_elements(),
+            fmp: fmp,
+            memory: mem.clone(),
```

Comment:
I haven't noticed this before, but this doesn't look right: memory gets updated at the next cycle. Here, it should still have been `Vec::new()`. Is it possible that we are somehow "off-by-one" in how we build memory state?